tf2_ros/src/static_transform_broadcaster.cpp

```diff
@@ -42,36 +43,22 @@ StaticTransformBroadcaster::StaticTransformBroadcaster()
   publisher_ = node_.advertise<tf2_msgs::TFMessage>("/tf_static", 100, true);
 };
 
-void StaticTransformBroadcaster::sendTransform(const geometry_msgs::TransformStamped & msgtf)
-{
-  std::vector<geometry_msgs::TransformStamped> v1;
-  v1.push_back(msgtf);
-  sendTransform(v1);
-}
-
-
 void StaticTransformBroadcaster::sendTransform(const std::vector<geometry_msgs::TransformStamped> & msgtf)
 {
-  for (std::vector<geometry_msgs::TransformStamped>::const_iterator it_in = msgtf.begin(); it_in != msgtf.end(); ++it_in)
+  for (const geometry_msgs::TransformStamped& input : msgtf)
```

Comment:
The two nested loops here kind of obscured the actual semantics of the code, namely iterating over all elements of `msgtf`, potentially finding an existing element, and updating it.
The proposed code explicitly states this semantics.